### ISSUE TITLE
[java] InsufficientStringBufferDeclaration should consider constant String. 

### ISSUE THREAD

**Issue by MetaBF** [pmd/pmd#3857]
**Affects PMD Version:** 

**Rule: InsufficientStringBufferDeclaration**
 https://pmd.github.io/latest/pmd_rules_java_performance.html#insufficientstringbufferdeclaration 

**Description:**

Rule **InsufficientStringBufferDeclaration** is supposed to prevent unnecessary resizing. In this case, the size of the constant variable `PRE` is 21 (at point 1). `PRE` is then appended to local variable `builder` (at point 3), a StringBuilder instance initialized with default capacity 16 (at point 2).  

Hence PMD should report a violation at line 10 (at point 2).

**Code Sample demonstrating the issue:**

```
import java.util.ArrayList;
import java.util.List;

public class Example {
    public List<String> list = new ArrayList<>();

    public static final String PRE = "The size of list is: "; // point 1: 21 characters, exceed default 16 

    public void func() {
        StringBuilder builder = new StringBuilder(); // point 2: improper initial capacity, false negative
        builder.append(PRE); // point 3
    }
}

```

**Expected outcome:**

PMD should report a violation at line 10, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
